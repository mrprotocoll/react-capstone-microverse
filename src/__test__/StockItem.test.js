const { render } = require('@testing-library/react');
const { default: Hero } = require('components/Hero');
const { Provider } = require('react-redux');
const { BrowserRouter } = require('react-router-dom');
const { default: store } = require('redux/store');

describe('StockItem Component', () => {
  test('component match the snapshot', () => {
    const data = [
      {
        symbol: 'KZMS.ME',
        name: 'The Open Joint Stock Company Krasnokamsk Metal Mesh Works',
        price: 226,
        exchange: 'MCX',
        exchangeShortName: 'MCX',
        type: 'stock',
      },
      {
        symbol: 'YAFFX',
        name: 'AMG Yacktman Focused Fund',
        price: 18.02,
        exchange: 'Nasdaq',
        exchangeShortName: 'NASDAQ',
        type: 'trust',
      },
      {
        symbol: 'YAFIX',
        name: 'AMG Funds - AMG Yacktman Focused Fund',
        price: 17.96,
        exchange: 'Nasdaq',
        exchangeShortName: 'NASDAQ',
        type: 'trust',
      },
      {
        symbol: 'ZDIVX',
        name: 'Zacks Dividend Fund Investor Class',
        price: 21.66,
        exchange: 'Nasdaq',
        exchangeShortName: 'NASDAQ',
        type: 'trust',
      },
    ];

    const stockItem = render(
      <Provider store={store}>
        <BrowserRouter>
          <Hero data={data} />
        </BrowserRouter>
      </Provider>,
    );
    expect(stockItem).toMatchSnapshot();
  });
});
