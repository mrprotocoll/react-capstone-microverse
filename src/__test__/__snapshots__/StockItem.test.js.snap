// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StockItem Component component match the snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="hero"
      >
        <div
          class="overlay"
        />
        <a
          class="content"
          href="/details/undefined"
        >
          <h2 />
          <small />
          <h3 />
        </a>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="hero"
    >
      <div
        class="overlay"
      />
      <a
        class="content"
        href="/details/undefined"
      >
        <h2 />
        <small />
        <h3 />
      </a>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
